src/openloco/objects/train_signal_object.h

```diff
@@ -16,11 +16,21 @@ namespace openloco
         uint8_t cost_index;        // 0x0A
         uint8_t var_0B;
         uint16_t var_0C;
-        uint32_t var_0E;
+        uint32_t image;
         uint8_t num_compatible; // 0x12
         uint8_t mods[7];        // 0x13
         uint16_t designed_year; // 0x1A
         uint16_t obsolete_year; // 0x1C
     };
 #pragma pack(pop)
+
+    const std::vector<uint8_t> signalFrames2State = { 1, 2, 3, 3, 3, 3, 3, 3, 2, 1, 0, 0, 0, 0, 0 };
```

Comment:
Use a std array for fixed sized things like this

Reply:
I can't do that due to the signalFrames vector below in L31

Reply:
Oh right its a jagged array.